File: example-bots/node/random_bot.mjs
// Minimal "random mover" Bomberman bot in Node.js (no dependencies).
// Run it with: node random_bot.mjs
// Then point an adapter at: http://<this-host>:8080/action

import http from "node:http";

const ACTIONS = ["up", "down", "left", "right", "bomb", "stay"];
const pick = (arr) => arr[Math.floor(Math.random() * arr.length)];

http
  .createServer((req, res) => {
    if (req.method !== "POST" || req.url !== "/action") {
      res.writeHead(404).end();
      return;
    }
    let raw = "";
    req.on("data", (chunk) => (raw += chunk));
    req.on("end", () => {
      const payload = JSON.parse(raw || "{}");
      const me = payload.you ?? "a";
      const unitIds = payload.game_state.agents[me].unit_ids;
      const actions = unitIds.map((unit_id) => ({
        unit_id,
        action: pick(ACTIONS),
      }));
      const body = JSON.stringify({ actions });
      res.writeHead(200, { "Content-Type": "application/json" }).end(body);
    });
  })
  .listen(8080);

File: example-bots/python/random_bot.py
"""Minimal "random mover" Bomberman bot. Stdlib only, ~40 lines.

Run it with: python3 random_bot.py
Then point an adapter at: http://<this-host>:8080/action
"""
import json
import random
from http.server import BaseHTTPRequestHandler, HTTPServer

ACTIONS = ["up", "down", "left", "right", "bomb", "stay"]


class Handler(BaseHTTPRequestHandler):
    def do_POST(self):  # noqa: N802
        length = int(self.headers.get("Content-Length", "0"))
        payload = json.loads(self.rfile.read(length) or b"{}")
        me = payload.get("you", "a")
        my_unit_ids = payload["game_state"]["agents"][me]["unit_ids"]

        actions = [
            {"unit_id": uid, "action": random.choice(ACTIONS)} for uid in my_unit_ids
        ]
        body = json.dumps({"actions": actions}).encode()

        self.send_response(200)
        self.send_header("Content-Type", "application/json")
        self.send_header("Content-Length", str(len(body)))
        self.end_headers()
        self.wfile.write(body)

    def log_message(self, fmt, *args):  # silence default access log
        pass


if __name__ == "__main__":
    HTTPServer(("0.0.0.0", 8080), Handler).serve_forever()
